Definicija/Definicija.cs

```diff
@@ -12,46 +12,66 @@ public Osoba(string ime, string prezime)
                 Prezime = prezime;
             }
 
-            // TODO:000 Javno dostupno polje Prezime nadomjestiti svojstvom (property) koje se izvan klase može samo čitati, a mijenjati se može samo iz klase.
+            // Javno dostupno polje Prezime nadomjestiti svojstvom (property) koje se izvan klase može samo čitati, a mijenjati se može samo iz klase.
 
-            public string Prezime;
+            public string Prezime { get;  set; }
 
 
-            // TODO:001 Javno dostupno polje Ime nadomjestiti svojstvom (property) koje se izvan klase može čitati i zadavati, ali prilikom zadavanja treba onemogućiti
+            //  Javno dostupno polje Ime nadomjestiti svojstvom (property) koje se izvan klase može čitati i zadavati, ali prilikom zadavanja treba onemogućiti
             // zadavanje nul-referencom ili praznim znakovnim nizom. U tim slučajevima treba baciti iznimke tipa ArgumentNullException, odnosno ArgumentException.
 
-            public string Ime;
+            public string Ime
+            {
+                get { return Ime; }
+                set
+                {
+                    if (value == null)
+                        throw new ArgumentNullException(nameof(value), $"Vrijednost {nameof(Ime)} ne smije biti nul referenca");
+                    if (value == string.Empty)
+                        throw new ArgumentException($"Vrijednost {nameof(Ime)} ne smije biti prazan string", nameof(value));
+                    Ime = value;
```

Comment:
Morate u klasu dodati privatno polje:
`private string ime;`
te unutar konstruktora i unutar svojstva Ime dohvaćati taj privatni član.
Trenutno se baca StackOverflowException jer u set i get metodama svojstva Ime ponovno rekurzivno pozivate to svojstvo.